lumicks/pylake/tests/test_kymo.py

```diff
@@ -213,3 +223,43 @@ def test_save_tiff(tmpdir_factory, h5_file):
         kymo = f.kymos["Kymo1"]
         kymo.save_tiff(f"{tmpdir}/kymo1.tiff")
         assert stat(f"{tmpdir}/kymo1.tiff").st_size > 0
+
+
+def test_downsampled_kymo():
+    image = np.array(
+        [
+            [0, 12, 0, 12, 0, 6, 0],
+            [0, 0, 0, 0, 0, 6, 0],
+            [12, 0, 0, 0, 12, 6, 0],
+            [0, 12, 12, 12, 0, 6, 0],
+        ],
+        dtype=np.uint8
+    )
+
+    kymo = generate_kymo(
+        "Mock", image, pixel_size_nm=1, start=100, dt=7, samples_per_pixel=5, line_padding=2
+    )
+
+    kymo_ds = kymo.downsampled_by(2)
+    ds = np.array(
+        [
+            [12, 12, 6],
+            [0, 0, 6],
+            [12, 0, 18],
+            [12, 24, 6],
+        ],
+        dtype=np.uint8
+    )
+
+    assert kymo_ds.name == "Mock"
+    assert np.allclose(kymo_ds.red_image, ds)
+    assert np.allclose(kymo_ds.start, 100)
+    assert np.allclose(kymo_ds.pixelsize_um, 1 / 1000)
+    assert np.allclose(kymo_ds.line_time_seconds, 2 * 7 * (5 * 4 + 2 + 2) / 1e9)
+
+    # Per pixel timestamps cannot reasonably be computed for a downsampled image
+    with pytest.raises(NotImplementedError):
+        kymo_ds.timestamps
```

Comment:
The explanation should be part of the error message. And then you can test for a matching message here. Then the test is self-documented and you don't need the comment.

Reply:
Good point. :)